Architecture.md

```diff
@@ -182,10 +189,65 @@ As described previously, the API manages most of the user interactions and redir
+
+As you can notice, every table has a UUID field as a primary key. Compared to [Serial](https://www.postgresql.org/docs/current/datatype-numeric.html) that is often used to identify a table, a UUID guarantees a better uniqueness across the whole database. 
+To generate a [UUID](https://www.postgresql.org/docs/current/datatype-uuid.html), we are using the UUIDv4 standard, which generates each UUID randomly. Thus, there are roughly 103 trillion UUIDv4s, lowering the chance of finding a duplicate UUID to one-in-a-billion. 
+In comparison, Serials are limited to roughly 2 billion (2147483647) and will eventually clog up the column and be reduntant. As the column is a primary key, this results in the table being full and creates an inability to insert new rows. 
+However, even though serials take up less space (4 bytes) than a UUID (16 bytes), the column becomes a series of incrementing integers and offers information about the time of creation. This makes the table prone to timestamp-guessing. UUIDs are generated randomly and impossible to guess, completely removing this attack pattern.
+
+
+Furthermore, since the projects will be deployed at a subdomain that is named after `{projectId}.user-app.paastech.cloud` we needed to hide the internal database structure that emerges with a serial id. Therefore, using a uuid prevents sharing sensitive data with everyone and prevents targeted attacks.
+
+
+The table `users` contains all the necessary information about each User. Upon User creation, an `email_nonce` is automatically created and a Client account is only considered active once the email has been confirmed and the field `email_nonce` is null.
+A nonce is a random value that is unique and serves for identification. Should the User wish to reset their password, a new UUID will be saved in the field `password_nonce`. To reset his password, the Client will need to provide said nonce together with the new password. Should another request to change the password been made before the password was reset, a new nonce will be generated and the old one will be overwritten. Once the password has been reset, `password_nonce` will be set to null. 
+One User can have multiple projects or SSH keys.
+
+The `ssh_keys` specified by each User allow them to push their repository onto our git server. Each SSH key can have a name to make it easier to distinguish multiple keys, however, it is not required.
+An SSH key belongs to a Client and not a Project. Thus, it will allow the Client to access any of their repositories. Should a User possess more than one SSH key, each one will grant them access.
+
+
+The `projects` table describes a Project. Its field `config` contains all the environmental variables of said Project, like a database URL. Since the configuration changes for every Project, we decided to store it as a flexible json field. We decided to use a JSONB field that stores the JSON data in binary form, allowing for better performances than a simple json field.
```

Comment:
I also spanned it across multiple lines (2 actually, but that's a 100% more !)

Wait while I'm at it, where do you back your performance claims about JSONB compared to JSON ?
```suggestion
The `projects` table describes a Project. Its `config` field contains all the environment variables of said Project, like a database URL. 
Since the configuration changes for every Project, we decided to store it as a flexible JSON field. We decided to use a JSONB field that stores the JSON data in binary form, allowing for better performance than a simple JSON field.
```